### ISSUE TITLE
ERROR: Assertion Failed

### ISSUE THREAD

**Issue by camdagr8** [phonegap/ios-deploy#11]
I get the following error whenever I try to deploy anything to a device: 

Assertion failed: (AMDeviceStartService(device, CFSTR("com.apple.debugserver"), &gdbfd, NULL) == 0), function start_remote_debug_server, file ios-deploy.c, line 524.

/Developer/Projects/Pathem/App/Platforms/ios/cordova/run: line 125: 15533 Abort trap: 6           ios-deploy -d -b "$DEVICE_APP_PATH"

I'm using the most recent version of ios-deploy and cordova (3.1.0).

**Comment by aogilvie** [comment]
@shazron I found I was getting this error when the path to the application was incorrect [`ios-deploy -d -b "$DEVICE_APP_PATH"`]

Running the following command manually with the .app file path allowed me push without issues to a device.

`ios-deploy --debug -d -b /Users/aogilvie/Documents/wizcorp/<appname>/platforms/ios/build/device/<appname>.app`

So I think `DEVICE_APP_PATH` is incorrect.

I could replicate an assertion error and throw "ios-deploy quit unexpectedly" as follows:
```
ios-deploy --debug -d -b /Users/aogilvie/Documents/wizcorp/<appname>/platforms/ios/build/<appname>.build/Debug-iphoneos/<appname>.build/<appname>.app.xcent /Users/aogilvie/Documents/wizcorp/<appname>/platforms/ios/build/device/<appname>.app
[....] Waiting for iOS device to be connected
[....] Using iPod touch 5G 'wizcorp_iPod5_001' (63f701b1a48d857f95bc022fb527a475f3356973) (63f701b1a48d857f95bc022fb527a475f3356973).
------ Install phase ------
[  0%] Found iPod touch 5G 'wizcorp_iPod5_001' (63f701b1a48d857f95bc022fb527a475f3356973) connected through USB, beginning install
Assertion failed: (AMDeviceSecureTransferPath(0, device, url, options, transfer_callback, 0)==0), function handle_device, file ios-deploy.c, line 1532.
Abort trap: 6
```

**Comment by Shilo** [comment]
I had the same issue. My fix was to allow my iOS device to "trust" my computer. It was easily overlooked when re-installing OS X.

**Comment by ohh2ahh** [comment]
I have the same issue with OSX Yosemite and ios-deploy v1.4.0. Killing old instance lldb didn't help.

```
$ ios-deploy --debug --bundle platforms/ios/build/device/<appname>.app
// or
$ cordova run ios --device
// or
$ ionic run ios --device
[....] Waiting for iOS device to be connected
[....] Using iPhone 6 (GSM) 'iPhone 6' (227eb7<...>) (227eb7<...>).
------ Install phase ------
[  0%] Found iPhone 6 (GSM) 'iPhone 6' (227eb7<...>) connected through USB, beginning install
Assertion failed: (AMDeviceSecureTransferPath(0, device, url, options, transfer_callback, 0)==0), function handle_device, file ios-deploy.c, line 1516.
Abort trap: 6
```

* **sw_vers:** 10.10.2
* **ios-deploy:** 1.4.0
* **xcodebuild:** Xcode 6.2, Build version 6C131e
* **xcode-select:** `/Applications/Xcode.app/Contents/Developer`
* **gcc:**
```
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk/usr/include/c++/4.2.1
Apple LLVM version 6.0 (clang-600.0.57) (based on LLVM 3.5svn)
Target: x86_64-apple-darwin14.1.0
Thread model: posix
```
* **lldb:** lldb-320.4.160

**Comment by shazron** [comment]
1.5.0 has some improvements, let me know. https://github.com/phonegap/ios-deploy/releases/tag/1.5.0

**Comment by listrophy** [comment]
I had been getting the same as @drastick: 

```
[ 0%] Looking up developer disk image
[ !! ] Unable to mount developer disk image. (e8000033)
```

I realized that I had a beta of Xcode selected through `xcode-select`, so a quick `sudo xcode-select -s /Applications/Xcode.app/Contents/Developer` got me through this.

Next, I got this:

```
[100%] Installed package /Users/path/to/app/platforms/ios/build/device/MyApp.app
------ Debug phase ------
Cannot debug (...1234udid...) over WIFI.
```

Strangely enough, I don't even have WiFi Sync enabled, according to iTunes. And yes, the device is plugged in through a simple lightning USB cable. The app was getting on the phone, but it wasn't getting run. So, next I deleted the app and tried again.

Now, I get this:

```
[100%] Installed package /Users/path/to/app/platforms/ios/build/device/MyApp.app
------ Debug phase ------
Starting debug of iPhone 5s (GSM) 'Juliette' (...123udid...) connected through USB...
[  0%] Looking up developer disk image
[ 95%] Developer disk image mounted successfully
/Users/path/to/app/platforms/ios/cordova/run: line 116: 16666 Abort trap: 6           ios-deploy -d -b "$DEVICE_APP_PATH"
```

Again, the app gets successfully deployed to the phone, but it doesn't get run.

My config:
```
$ sw_vers -productVersion
10.10.3
$ ios-deploy -V
1.5.0
$ xcodebuild -version
Xcode 6.3
Build version 6D570
$ xcode-select --print-path
/Applications/Xcode.app/Contents/Developer
$ gcc --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 6.1.0 (clang-602.0.49) (based on LLVM 3.6.0svn)
Target: x86_64-apple-darwin14.3.0
Thread model: posix
$ lldb --version
lldb-330.0.44
```

Apologies if this is actually a cordova or ionic issue, but I'm thinking it's ios-deploy.

**Comment by shazron** [comment]
@listrophy you mentioned that it is installed but doesn't run. Can it run manually? (just checking the integrity of the install)

**Comment by shazron** [comment]
@listrophy, in the root of your cordova app, can you run:
```
cordova build --device 
ios-deploy -d -b platforms/ios/build/device/YOURAPPNAME.app
```
Substitute YOURAPPNAME for your app's name. If you get an error, paste it here.

**Comment by ohh2ahh** [comment]
I think that I've finally found the cause of [my issue](https://github.com/phonegap/ios-deploy/issues/11#issuecomment-85138881). I'm using a german umlaut (`ü`) in my app name (configured in `config.xml'), which is obviously not supported. I'd be very happy if you could fix this.

**Comment by shazron** [comment]
@ohh2ahh thanks, I'll file a new issue. Just to confirm, if you just use ascii letters in your app name it works fine, yes?

**Comment by shazron** [comment]
Filed #147

**Comment by ohh2ahh** [comment]
@shazron Yes, without the umlaut everything works perfectly fine. Thanks for your effort, I appreciate it very much!

**Comment by haxpor** [comment]
@listrophy I have the same issue with iOS 8.4 and Xcode 8.4. Just shutting down your phone and restart again will fix the problem in my case. See https://github.com/phonegap/ios-deploy/issues/155.

**Comment by shazron** [comment]
Fixed in #56
